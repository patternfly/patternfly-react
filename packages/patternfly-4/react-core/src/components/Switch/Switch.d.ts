import { FormEvent, FunctionComponent, HTMLProps } from 'react';
import { Omit } from '../../helpers/typeUtils';

export interface SwitchProps extends Omit<HTMLProps<HTMLInputElement>, 'type' | 'onChange' | 'disabled' | 'label'> {
  isDisabled?: boolean;
  isChecked?: boolean;
  onChange?(checked: boolean, event: FormEvent<HTMLInputElement>): void;
  id?: string;
  'aria-label': string;
  label?: string;
  className?: string;
}

declare const Switch: FunctionComponent<SwitchProps>;

export default Switch;
