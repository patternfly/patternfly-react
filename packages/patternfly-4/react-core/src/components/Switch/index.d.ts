export { default as Switch, SwitchProps } from './Switch';
