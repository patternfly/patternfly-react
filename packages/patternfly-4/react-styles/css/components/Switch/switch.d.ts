import './switch.css';
declare const _default: {
    switch: string;
    switchInput: string;
    switchLabel: string;
    switchToggle: string;
    switchToggleIcon: string;
    modifiers: {
        'off': string;
        'on': string;
    };
};
export default _default;
