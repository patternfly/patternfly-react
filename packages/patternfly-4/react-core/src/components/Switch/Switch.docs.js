import { Switch } from '@patternfly/react-core';
import SimpleSwitch from './examples/SimpleSwitch';
import NoLabelSwitch from './examples/NoLabelSwitch';
import DisabledSwitch from './examples/DisabledSwitch';
import UncontrolledSwitch from './examples/UncontrolledSwitch';

export default {
  title: 'Switch',
  components: {
    Switch
  },
  examples: [
    { component: SimpleSwitch, title: 'Simple Switch' },
    { component: NoLabelSwitch, title: 'Switch with no labels' },
    { component: DisabledSwitch, title: 'Disabled Switch' },
    { component: UncontrolledSwitch, title: 'Uncontrolled Switch' }
  ]
};
