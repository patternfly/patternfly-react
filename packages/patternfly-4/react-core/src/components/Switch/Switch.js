import React from 'react';
import PropTypes from 'prop-types';
import styles from '@patternfly/react-styles/css/components/Switch/switch';
import { css } from '@patternfly/react-styles';
import { CheckIcon } from '@patternfly/react-icons';
import { getUniqueId } from '../../helpers/util';

const propTypes = {
  /** id for the label. */
  id: PropTypes.string,
  /** Additional classes added to the Switch */
  className: PropTypes.string,
  /** Text value for the label */
  label: PropTypes.string,
  /** Flag to show if the Switch is checked. */
  isChecked: PropTypes.bool,
  /** Flag to show if the Switch is disabled. */
  isDisabled: PropTypes.bool,
  /** A callback for when the Switch selection changes. (isChecked, event) => {} */
  onChange: PropTypes.func,
  /** Adds accessible text to the Switch, and should describe the isChecked="true" state. When label is defined, aria-label should be set to the text string that is visible when isChecked is true. */
  'aria-label': props => {
    if (!props.id && !props['aria-label']) {
      return new Error('Switch requires either an id or aria-label to be specified');
    }
    return null;
  },
  /** Additional props are spread to the container <input> */
  '': PropTypes.any // eslint-disable-line react/require-default-props
};

const defaultProps = {
  id: '',
  className: '',
  label: '',
  isChecked: true,
  isDisabled: false,
  onChange: () => undefined,
  'aria-label': ''
};

class Switch extends React.Component {
  id = this.props.id || getUniqueId();

  render() {
    const { id, className, label, isChecked, isDisabled, onChange, ...props } = this.props;
    return (
      <label className={css(styles.switch, className)} htmlFor={this.id}>
        <input
          {...props}
          id={this.id}
          className={css(styles.switchInput)}
          type="checkbox"
          onChange={event => onChange(event.currentTarget.checked, event)}
          checked={isChecked}
          disabled={isDisabled}
        />
        {label !== '' ? (
          <React.Fragment>
            <span className={css(styles.switchToggle)} />
            <span className={css(styles.switchLabel, styles.modifiers.on)} aria-hidden="true">
              {label}
            </span>
            <span className={css(styles.switchLabel, styles.modifiers.off)} aria-hidden="true">
              {label}
            </span>
          </React.Fragment>
        ) : (
          <span className={css(styles.switchToggle)}>
            <div className={css(styles.switchToggleIcon)} aria-hidden="true">
              <CheckIcon noVerticalAlign />
            </div>
          </span>
        )}
      </label>
    );
  }
}

Switch.propTypes = propTypes;
Switch.defaultProps = defaultProps;

export default Switch;
