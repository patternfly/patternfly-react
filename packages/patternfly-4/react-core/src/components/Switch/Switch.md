---
title: 'Switch'
cssPrefix: 'pf-c-switch'
propComponents: ['Switch']
---

import { Switch } from '@patternfly/react-core';

## Simple switch
```js
import React from 'react';
import { Switch } from '@patternfly/react-core';

class SimpleSwitch extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      isChecked: true
    };
    this.handleChange = isChecked => {
      this.setState({ isChecked });
    };
  }

  render() {
    const { isChecked } = this.state;
    return (
      <Switch
        id="simple-switch"
        label={isChecked ? 'Message when on' : 'Message when off'}
        isChecked={isChecked}
        onChange={this.handleChange}
        aria-label="Message when on"
      />
    );
  }
}
```

## Switch (no labels)
```js
import React from 'react';
import { Switch } from '@patternfly/react-core';

class NoLabelSwitch extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      isChecked: true
    };
    this.handleChange = isChecked => {
      this.setState({ isChecked });
    };
  }

  render() {
    const { isChecked } = this.state;
    return (
      <Switch id="no-label-switch-on" aria-label="Message when on" isChecked={isChecked} onChange={this.handleChange} />
    );
  }
}
```

## Switch (disabled)
```js
import React from 'react';
import { Switch } from '@patternfly/react-core';

DisabledSwitch = () => (
  <React.Fragment>
    <Switch id="disabled-switch-on" aria-label="Message when on" label="Message when on" isChecked isDisabled />
    <br />
    <Switch
      id="disabled-switch-off"
      aria-label="Message when on"
      label="Message when off"
      isChecked={false}
      isDisabled
    />
    <br />
    <Switch id="disabled-no-label-switch-on" aria-label="Message when on" isChecked isDisabled />
    <br />
    <Switch id="disabled-no-label-switch-off" aria-label="Message when on" isChecked={false} isDisabled />
  </React.Fragment>
);
```

## Switch (uncontrolled)
```js
import React from 'react';
import { Switch } from '@patternfly/react-core';

UncontrolledSwitch = () => (
  <React.Fragment>
    <Switch id="uncontrolled-switch-on" aria-label="Message when on" label="Message when on" isChecked />
    <br />
    <Switch id="uncontrolled-switch-off" aria-label="Message when on" label="Message when off" isChecked={false} />
    <br />
    <Switch id="uncontrolled-no-label-switch-on" aria-label="Message when on" isChecked />
    <br />
    <Switch id="uncontrolled-no-label-switch-off" aria-label="Message when on" isChecked={false} />
  </React.Fragment>
);
```
