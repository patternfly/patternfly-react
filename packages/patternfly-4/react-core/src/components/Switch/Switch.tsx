import * as React from 'react';
import styles from '@patternfly/react-styles/css/components/Switch/switch';
import { css } from '@patternfly/react-styles';
import { CheckIcon } from '@patternfly/react-icons';
import { getUniqueId } from '../../helpers/util';
import { Omit } from '../../helpers/typeUtils';
import { isOUIAEnvironment, getUniqueId as getOUIAUniqueId } from '../../helpers/ouia';

export interface SwitchProps extends Omit<React.HTMLProps<HTMLInputElement>, 'type' | 'onChange' | 'disabled' | 'label'> {
  /** id for the label. */
  id?: string,
  /** Additional classes added to the Switch */
  className?: string,
  /** Text value for the label */
  label?: string,
  /** Flag to show if the Switch is checked. */
  isChecked?: boolean,
  /** Flag to show if the Switch is disabled. */
  isDisabled?: boolean,
  /** A callback for when the Switch selection changes. (isChecked, event) => {} */
  onChange?: (checked: boolean, event: React.FormEvent<HTMLInputElement>) => void;
  /** Adds accessible text to the Switch, and should describe the isChecked="true" state. When label is defined, aria-label should be set to the text string that is visible when isChecked is true. */
  'aria-label'?: string
};

export class Switch extends React.Component<SwitchProps> {
  id = '';
  ouiaId = getOUIAUniqueId();

  static defaultProps = {
    id: '',
    className: '',
    label: '',
    isChecked: true,
    isDisabled: false,
    'aria-label': '',
    onChange: () => undefined as any
  };

  constructor(props: SwitchProps) {
    super(props);
    if (!props.id && !props['aria-label']) {
      // tslint:disable-next-line:no-console
      console.error('Switch: Switch requires either an id or aria-label to be specified');
    }
    this.id =props.id || getUniqueId();
  }

  render() {
    const { className, label, isChecked, isDisabled, onChange, ...props } = this.props;
    return (
      <label
        className={css(styles.switch, className)}
        htmlFor={this.id}
        {...isOUIAEnvironment() && {
          'data-ouia-component-type': 'Switch',
          'data-ouia-component-id': this.ouiaId
        }}
      >
        <input
          {...props}
          id={this.id}
          className={css(styles.switchInput)}
          type="checkbox"
          onChange={event => onChange(event.currentTarget.checked, event)}
          checked={isChecked}
          disabled={isDisabled}
        />
        {label !== '' ? (
          <React.Fragment>
            <span className={css(styles.switchToggle)} />
            <span className={css(styles.switchLabel, styles.modifiers.on)} aria-hidden="true">
              {label}
            </span>
            <span className={css(styles.switchLabel, styles.modifiers.off)} aria-hidden="true">
              {label}
            </span>
          </React.Fragment>
        ) : (
          <span className={css(styles.switchToggle)}>
            <div className={css(styles.switchToggleIcon)} aria-hidden="true">
              <CheckIcon noVerticalAlign />
            </div>
          </span>
        )}
      </label>
    );
  }
}
