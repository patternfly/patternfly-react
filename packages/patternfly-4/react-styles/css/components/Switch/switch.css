/* stylelint-enable */
/* stylelint-disable */
/* stylelint-enable */
.pf-c-switch {
  --pf-c-switch--Height: calc(var(--pf-global--FontSize--md) * var(--pf-global--LineHeight--md));
  --pf-c-switch--Width: calc(var(--pf-c-switch--Height) * var(--pf-global--golden-ratio));
  --pf-c-switch--LineHeight: var(--pf-global--LineHeight--md);
  --pf-c-switch__input--checked__toggle--BackgroundColor: var(--pf-global--primary-color--100);
  --pf-c-switch__input--checked__toggle--before--Transform: translateX(calc(var(--pf-c-switch--Width) - var(--pf-c-switch__toggle--before--Width) - 2 * var(--pf-c-switch__toggle--before--Left)));
  --pf-c-switch__input--checked__label--Color: var(--pf-global--Color--dark-100);
  --pf-c-switch__input--not-checked__label--Color: var(--pf-global--disabled-color--100);
  --pf-c-switch__input--disabled__label--Color: var(--pf-global--disabled-color--100);
  --pf-c-switch__input--disabled__toggle--BackgroundColor: var(--pf-global--Color--dark-200);
  --pf-c-switch__input--disabled__toggle--before--BackgroundColor: var(--pf-global--disabled-color--200);
  --pf-c-switch__toggle--BackgroundColor: var(--pf-global--disabled-color--200);
  --pf-c-switch__toggle--BorderRadius: var(--pf-c-switch--Height);
  --pf-c-switch__toggle--before--Top: calc(var(--pf-c-switch--Height) / 2 * (1 - 1 / var(--pf-global--golden-ratio)));
  --pf-c-switch__toggle--before--Left: var(--pf-c-switch__toggle--before--Top);
  --pf-c-switch__toggle--before--Width: calc(var(--pf-c-switch--Height) / var(--pf-global--golden-ratio));
  --pf-c-switch__toggle--before--Height: var(--pf-c-switch__toggle--before--Width);
  --pf-c-switch__toggle--before--BackgroundColor: var(--pf-global--BackgroundColor--100);
  --pf-c-switch__toggle--before--BorderRadius: var(--pf-global--BorderRadius--lg);
  --pf-c-switch__toggle--before--BoxShadow: var(--pf-global--BoxShadow--md);
  --pf-c-switch__toggle-icon--FontSize: var(--pf-global--icon--FontSize--sm);
  --pf-c-switch__toggle-icon--Color: var(--pf-global--Color--light-100);
  --pf-c-switch__toggle-icon--PaddingLeft: var(--pf-global--spacer--sm);
  --pf-c-switch__toggle-icon--Top: 0.4375rem;
  --pf-c-switch__label--PaddingLeft: var(--pf-global--spacer--md);
  --pf-c-switch__label--FontSize: var(--pf-global--FontSize--md);
  --pf-c-switch__label--FontWeight: var(--pf-global--FontWeight--normal);
  --pf-c-switch__label--LineHeight: var(--pf-global--LineHeight--md);
  --pf-c-switch__label--Color: var(--pf-global--Color--dark-100);
  position: relative;
  display: inline-block;
  height: var(--pf-c-switch--Height);
  line-height: var(--pf-c-switch--LineHeight);
  vertical-align: middle;
  cursor: pointer;
  /* stylelint-enable */ }
  .pf-c-switch .pf-c-switch__input {
    position: absolute;
    cursor: pointer;
    opacity: 0;
    /* stylelint-disable */ }
    .pf-c-switch .pf-c-switch__input:checked ~ .pf-c-switch__label {
      color: var(--pf-c-switch__input--checked__label--Color); }
    .pf-c-switch .pf-c-switch__input:checked ~ .pf-c-switch__toggle {
      background-color: var(--pf-c-switch__input--checked__toggle--BackgroundColor); }
      .pf-c-switch .pf-c-switch__input:checked ~ .pf-c-switch__toggle::before {
        transform: var(--pf-c-switch__input--checked__toggle--before--Transform); }
    .pf-c-switch .pf-c-switch__input:checked ~ .pf-m-off {
      display: none; }
    .pf-c-switch .pf-c-switch__input:not(:checked) ~ .pf-c-switch__label {
      color: var(--pf-c-switch__input--not-checked__label--Color); }
    .pf-c-switch .pf-c-switch__input:not(:checked) ~ .pf-c-switch__toggle .pf-c-switch__toggle-icon {
      display: none;
      visibility: hidden; }
    .pf-c-switch .pf-c-switch__input:not(:checked) ~ .pf-m-on {
      display: none; }
    .pf-c-switch .pf-c-switch__input:disabled {
      cursor: not-allowed; }
      .pf-c-switch .pf-c-switch__input:disabled ~ .pf-c-switch__label {
        color: var(--pf-c-switch__input--disabled__label--Color);
        cursor: not-allowed; }
      .pf-c-switch .pf-c-switch__input:disabled ~ .pf-c-switch__toggle {
        cursor: not-allowed;
        background-color: var(--pf-c-switch__input--disabled__toggle--BackgroundColor); }
        .pf-c-switch .pf-c-switch__input:disabled ~ .pf-c-switch__toggle::before {
          background-color: var(--pf-c-switch__input--disabled__toggle--before--BackgroundColor); }
  .pf-c-switch .pf-c-switch__toggle {
    position: relative;
    display: inline-block;
    width: var(--pf-c-switch--Width);
    height: var(--pf-c-switch--Height);
    background-color: var(--pf-c-switch__toggle--BackgroundColor);
    border-radius: var(--pf-c-switch__toggle--BorderRadius); }
    .pf-c-switch .pf-c-switch__toggle::before {
      position: absolute;
      top: var(--pf-c-switch__toggle--before--Top);
      left: var(--pf-c-switch__toggle--before--Left);
      display: block;
      width: var(--pf-c-switch__toggle--before--Width);
      height: var(--pf-c-switch__toggle--before--Height);
      content: "";
      background-color: var(--pf-c-switch__toggle--before--BackgroundColor);
      border-radius: var(--pf-c-switch__toggle--before--BorderRadius);
      box-shadow: var(--pf-c-switch__toggle--before--BoxShadow);
      transition: transform .25s ease 0s, width .25s ease 0s, left .25s ease 0s;
      transform: translateX(0); }
  .pf-c-switch .pf-c-switch__toggle-icon {
    position: absolute;
    top: var(--pf-c-switch__toggle-icon--Top);
    padding-left: var(--pf-c-switch__toggle-icon--PaddingLeft);
    font-size: var(--pf-c-switch__toggle-icon--FontSize);
    color: var(--pf-c-switch__toggle-icon--Color); }
  .pf-c-switch .pf-c-switch__label {
    display: inline-block;
    padding-left: var(--pf-c-switch__label--PaddingLeft);
    font-size: var(--pf-c-switch__label--FontSize);
    font-weight: var(--pf-c-switch__label--FontWeight);
    line-height: var(--pf-c-switch__label--LineHeight);
    color: var(--pf-c-switch__label--Color);
    vertical-align: top; }
