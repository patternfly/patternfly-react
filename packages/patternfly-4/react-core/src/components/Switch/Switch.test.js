import React from 'react';
import { mount } from 'enzyme';
import { Switch } from './Switch';

const props = {
  onChange: jest.fn(),
  checked: false
};

test('switch label for attribute equals input id attribute', () => {
  const view = mount(<Switch id="foo" />);
  expect(view.find('input').prop('id')).toBe('foo');
  expect(view.find('label').prop('htmlFor')).toBe('foo');
});

test('switch label id is auto generated', () => {
  const view = mount(<Switch aria-label="..." />);
  expect(view.find('input').prop('id')).toBeDefined();
});

test('switch is checked', () => {
  const view = mount(<Switch id="switch-is-checked" label="On" labelOff="Off" isChecked />);
  expect(view).toMatchSnapshot();
});

test('switch is not checked', () => {
  const view = mount(<Switch id="switch-is-not-checked" label="On" labelOff="Off" isChecked={false} />);
  expect(view).toMatchSnapshot();
});

test('no label switch is checked', () => {
  const view = mount(<Switch id="no-label-switch-is-checked" isChecked />);
  expect(view).toMatchSnapshot();
});

test('no label switch is not checked', () => {
  const view = mount(<Switch id="no-label-switch-is-not-checked" isChecked={false} />);
  expect(view).toMatchSnapshot();
});

test('switch is checked and disabled', () => {
  const view = mount(<Switch id="switch-is-checked-and-disabled" isChecked isDisabled />);
  expect(view).toMatchSnapshot();
});

test('switch is not checked and disabled', () => {
  const view = mount(<Switch id="switch-is-not-checked-and-disabled" isChecked={false} isDisabled />);
  expect(view).toMatchSnapshot();
});

test('switch passes value and event to onChange handler', () => {
  const view = mount(<Switch id="onChange-switch" {...props} />);
  view.find('input').simulate('change', { target: { checked: true } });
  expect(view.find('input').prop('checked')).toBe(true);
});
