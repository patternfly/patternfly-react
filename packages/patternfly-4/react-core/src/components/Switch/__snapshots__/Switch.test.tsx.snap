// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`no label switch is checked 1`] = `
<Component
  id="no-label-switch-is-checked"
  isChecked={true}
>
  <ComponentWithOuia
    component={[Function]}
    componentProps={
      Object {
        "id": "no-label-switch-is-checked",
        "isChecked": true,
      }
    }
    consumerContext={null}
  >
    <Switch
      aria-label=""
      className=""
      id="no-label-switch-is-checked"
      isChecked={true}
      isDisabled={false}
      label=""
      labelOff=""
      onChange={[Function]}
      ouiaContext={
        Object {
          "isOuia": false,
          "ouiaId": null,
        }
      }
    >
      <label
        className="pf-c-switch"
        htmlFor="no-label-switch-is-checked"
      >
        <input
          aria-label=""
          aria-labelledby="no-label-switch-is-checked-on"
          checked={true}
          className="pf-c-switch__input"
          disabled={false}
          id="no-label-switch-is-checked"
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="pf-c-switch__toggle"
        >
          <div
            aria-hidden="true"
            className="pf-c-switch__toggle-icon"
          >
            <CheckIcon
              color="currentColor"
              noVerticalAlign={true}
              size="sm"
              title={null}
            >
              <svg
                aria-hidden={true}
                aria-labelledby={null}
                fill="currentColor"
                height="1em"
                role="img"
                style={null}
                viewBox="0 0 512 512"
                width="1em"
              >
                <path
                  d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
                  transform=""
                />
              </svg>
            </CheckIcon>
          </div>
        </span>
      </label>
    </Switch>
  </ComponentWithOuia>
</Component>
`;

exports[`no label switch is not checked 1`] = `
<Component
  id="no-label-switch-is-not-checked"
  isChecked={false}
>
  <ComponentWithOuia
    component={[Function]}
    componentProps={
      Object {
        "id": "no-label-switch-is-not-checked",
        "isChecked": false,
      }
    }
    consumerContext={null}
  >
    <Switch
      aria-label=""
      className=""
      id="no-label-switch-is-not-checked"
      isChecked={false}
      isDisabled={false}
      label=""
      labelOff=""
      onChange={[Function]}
      ouiaContext={
        Object {
          "isOuia": false,
          "ouiaId": null,
        }
      }
    >
      <label
        className="pf-c-switch"
        htmlFor="no-label-switch-is-not-checked"
      >
        <input
          aria-label=""
          aria-labelledby="no-label-switch-is-not-checked-on"
          checked={false}
          className="pf-c-switch__input"
          disabled={false}
          id="no-label-switch-is-not-checked"
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="pf-c-switch__toggle"
        >
          <div
            aria-hidden="true"
            className="pf-c-switch__toggle-icon"
          >
            <CheckIcon
              color="currentColor"
              noVerticalAlign={true}
              size="sm"
              title={null}
            >
              <svg
                aria-hidden={true}
                aria-labelledby={null}
                fill="currentColor"
                height="1em"
                role="img"
                style={null}
                viewBox="0 0 512 512"
                width="1em"
              >
                <path
                  d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
                  transform=""
                />
              </svg>
            </CheckIcon>
          </div>
        </span>
      </label>
    </Switch>
  </ComponentWithOuia>
</Component>
`;

exports[`switch is checked 1`] = `
<Component
  id="switch-is-checked"
  isChecked={true}
  label="On"
  labelOff="Off"
>
  <ComponentWithOuia
    component={[Function]}
    componentProps={
      Object {
        "id": "switch-is-checked",
        "isChecked": true,
        "label": "On",
        "labelOff": "Off",
      }
    }
    consumerContext={null}
  >
    <Switch
      aria-label=""
      className=""
      id="switch-is-checked"
      isChecked={true}
      isDisabled={false}
      label="On"
      labelOff="Off"
      onChange={[Function]}
      ouiaContext={
        Object {
          "isOuia": false,
          "ouiaId": null,
        }
      }
    >
      <label
        className="pf-c-switch"
        htmlFor="switch-is-checked"
      >
        <input
          aria-label=""
          aria-labelledby="switch-is-checked-on"
          checked={true}
          className="pf-c-switch__input"
          disabled={false}
          id="switch-is-checked"
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="pf-c-switch__toggle"
        />
        <span
          aria-hidden="true"
          className="pf-c-switch__label pf-m-on"
          id="switch-is-checked-on"
        >
          On
        </span>
        <span
          aria-hidden="true"
          className="pf-c-switch__label pf-m-off"
          id="switch-is-checked-off"
        >
          Off
        </span>
      </label>
    </Switch>
  </ComponentWithOuia>
</Component>
`;

exports[`switch is checked and disabled 1`] = `
<Component
  id="switch-is-checked-and-disabled"
  isChecked={true}
  isDisabled={true}
>
  <ComponentWithOuia
    component={[Function]}
    componentProps={
      Object {
        "id": "switch-is-checked-and-disabled",
        "isChecked": true,
        "isDisabled": true,
      }
    }
    consumerContext={null}
  >
    <Switch
      aria-label=""
      className=""
      id="switch-is-checked-and-disabled"
      isChecked={true}
      isDisabled={true}
      label=""
      labelOff=""
      onChange={[Function]}
      ouiaContext={
        Object {
          "isOuia": false,
          "ouiaId": null,
        }
      }
    >
      <label
        className="pf-c-switch"
        htmlFor="switch-is-checked-and-disabled"
      >
        <input
          aria-label=""
          aria-labelledby="switch-is-checked-and-disabled-on"
          checked={true}
          className="pf-c-switch__input"
          disabled={true}
          id="switch-is-checked-and-disabled"
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="pf-c-switch__toggle"
        >
          <div
            aria-hidden="true"
            className="pf-c-switch__toggle-icon"
          >
            <CheckIcon
              color="currentColor"
              noVerticalAlign={true}
              size="sm"
              title={null}
            >
              <svg
                aria-hidden={true}
                aria-labelledby={null}
                fill="currentColor"
                height="1em"
                role="img"
                style={null}
                viewBox="0 0 512 512"
                width="1em"
              >
                <path
                  d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
                  transform=""
                />
              </svg>
            </CheckIcon>
          </div>
        </span>
      </label>
    </Switch>
  </ComponentWithOuia>
</Component>
`;

exports[`switch is not checked 1`] = `
<Component
  id="switch-is-not-checked"
  isChecked={false}
  label="On"
  labelOff="Off"
>
  <ComponentWithOuia
    component={[Function]}
    componentProps={
      Object {
        "id": "switch-is-not-checked",
        "isChecked": false,
        "label": "On",
        "labelOff": "Off",
      }
    }
    consumerContext={null}
  >
    <Switch
      aria-label=""
      className=""
      id="switch-is-not-checked"
      isChecked={false}
      isDisabled={false}
      label="On"
      labelOff="Off"
      onChange={[Function]}
      ouiaContext={
        Object {
          "isOuia": false,
          "ouiaId": null,
        }
      }
    >
      <label
        className="pf-c-switch"
        htmlFor="switch-is-not-checked"
      >
        <input
          aria-label=""
          aria-labelledby="switch-is-not-checked-on"
          checked={false}
          className="pf-c-switch__input"
          disabled={false}
          id="switch-is-not-checked"
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="pf-c-switch__toggle"
        />
        <span
          aria-hidden="true"
          className="pf-c-switch__label pf-m-on"
          id="switch-is-not-checked-on"
        >
          On
        </span>
        <span
          aria-hidden="true"
          className="pf-c-switch__label pf-m-off"
          id="switch-is-not-checked-off"
        >
          Off
        </span>
      </label>
    </Switch>
  </ComponentWithOuia>
</Component>
`;

exports[`switch is not checked and disabled 1`] = `
<Component
  id="switch-is-not-checked-and-disabled"
  isChecked={false}
  isDisabled={true}
>
  <ComponentWithOuia
    component={[Function]}
    componentProps={
      Object {
        "id": "switch-is-not-checked-and-disabled",
        "isChecked": false,
        "isDisabled": true,
      }
    }
    consumerContext={null}
  >
    <Switch
      aria-label=""
      className=""
      id="switch-is-not-checked-and-disabled"
      isChecked={false}
      isDisabled={true}
      label=""
      labelOff=""
      onChange={[Function]}
      ouiaContext={
        Object {
          "isOuia": false,
          "ouiaId": null,
        }
      }
    >
      <label
        className="pf-c-switch"
        htmlFor="switch-is-not-checked-and-disabled"
      >
        <input
          aria-label=""
          aria-labelledby="switch-is-not-checked-and-disabled-on"
          checked={false}
          className="pf-c-switch__input"
          disabled={true}
          id="switch-is-not-checked-and-disabled"
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="pf-c-switch__toggle"
        >
          <div
            aria-hidden="true"
            className="pf-c-switch__toggle-icon"
          >
            <CheckIcon
              color="currentColor"
              noVerticalAlign={true}
              size="sm"
              title={null}
            >
              <svg
                aria-hidden={true}
                aria-labelledby={null}
                fill="currentColor"
                height="1em"
                role="img"
                style={null}
                viewBox="0 0 512 512"
                width="1em"
              >
                <path
                  d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
                  transform=""
                />
              </svg>
            </CheckIcon>
          </div>
        </span>
      </label>
    </Switch>
  </ComponentWithOuia>
</Component>
`;

exports[`switch with only label is checked 1`] = `
<Component
  id="switch-is-checked"
  isChecked={true}
  label="On"
>
  <ComponentWithOuia
    component={[Function]}
    componentProps={
      Object {
        "id": "switch-is-checked",
        "isChecked": true,
        "label": "On",
      }
    }
    consumerContext={null}
  >
    <Switch
      aria-label=""
      className=""
      id="switch-is-checked"
      isChecked={true}
      isDisabled={false}
      label="On"
      labelOff=""
      onChange={[Function]}
      ouiaContext={
        Object {
          "isOuia": false,
          "ouiaId": null,
        }
      }
    >
      <label
        className="pf-c-switch"
        htmlFor="switch-is-checked"
      >
        <input
          aria-label=""
          aria-labelledby="switch-is-checked-on"
          checked={true}
          className="pf-c-switch__input"
          disabled={false}
          id="switch-is-checked"
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="pf-c-switch__toggle"
        />
        <span
          aria-hidden="true"
          className="pf-c-switch__label pf-m-on"
          id="switch-is-checked-on"
        >
          On
        </span>
        <span
          aria-hidden="true"
          className="pf-c-switch__label pf-m-off"
          id="switch-is-checked-off"
        >
          On
        </span>
      </label>
    </Switch>
  </ComponentWithOuia>
</Component>
`;

exports[`switch with only label is not checked 1`] = `
<Component
  id="switch-is-not-checked"
  isChecked={false}
  label="Off"
>
  <ComponentWithOuia
    component={[Function]}
    componentProps={
      Object {
        "id": "switch-is-not-checked",
        "isChecked": false,
        "label": "Off",
      }
    }
    consumerContext={null}
  >
    <Switch
      aria-label=""
      className=""
      id="switch-is-not-checked"
      isChecked={false}
      isDisabled={false}
      label="Off"
      labelOff=""
      onChange={[Function]}
      ouiaContext={
        Object {
          "isOuia": false,
          "ouiaId": null,
        }
      }
    >
      <label
        className="pf-c-switch"
        htmlFor="switch-is-not-checked"
      >
        <input
          aria-label=""
          aria-labelledby="switch-is-not-checked-on"
          checked={false}
          className="pf-c-switch__input"
          disabled={false}
          id="switch-is-not-checked"
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="pf-c-switch__toggle"
        />
        <span
          aria-hidden="true"
          className="pf-c-switch__label pf-m-on"
          id="switch-is-not-checked-on"
        >
          Off
        </span>
        <span
          aria-hidden="true"
          className="pf-c-switch__label pf-m-off"
          id="switch-is-not-checked-off"
        >
          Off
        </span>
      </label>
    </Switch>
  </ComponentWithOuia>
</Component>
`;
